pygmt/tests/test_alias_system.py

```diff
@@ -0,0 +1,90 @@
+"""
+Tests for the alias system.
+"""
+
+import pytest
+from pygmt.alias import Alias, AliasSystem
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import build_arg_list
+
+
+def func(projection=None, region=None, frame=None, label=None, text=None, **kwargs):
+    """
+    A simple function to test the alias system.
+    """
+    alias = AliasSystem(
+        J=Alias(projection, name="projection"),
+        R=Alias(region, name="region", separator="/", size=[4, 6]),
+        B=Alias(frame, name="frame"),
+        U=[Alias(label, name="label"), Alias(text, name="text", prefix="+t")],
+    ).update(kwargs)
+    return build_arg_list(alias.kwdict)
+
+
+def test_alias_system_long_form():
+    """
+    Test that the alias system works with long-form parameters.
+    """
+    # One parameter
+    assert func(projection="X10c") == ["-JX10c"]
+    # Multiple parameters.
+    assert func(projection="H10c", region=[0, 10, 0, 20]) == ["-JH10c", "-R0/10/0/20"]
+    # Repeatable parameters.
+    assert func(frame=["WSen", "xaf", "yaf"]) == ["-BWSen", "-Bxaf", "-Byaf"]
+    # Multiple long-form parameters.
+    assert func(label="abcd", text="efg") == ["-Uabcd+tefg"]
+    assert func(
+        projection="H10c",
+        region=[0, 10, 0, 20],
+        label="abcd",
+        text="efg",
+        frame=["WSen", "xaf", "yaf"],
+    ) == ["-BWSen", "-Bxaf", "-Byaf", "-JH10c", "-R0/10/0/20", "-Uabcd+tefg"]
+
+
+def test_alias_system_one_alias_short_form():
+    """
+    Test that the alias system works when short-form parameters coexist.
+    """
+    # Long-form does not exist.
+    assert func(A="abc") == ["-Aabc"]
+
+    # Long-form exists but is not given, and short-form is given.
+    with pytest.warns(
+        SyntaxWarning,
+        match="Short-form parameter 'J' is not recommended. Use long-form parameter 'projection' instead.",
+    ):
+        assert func(J="X10c") == ["-JX10c"]
+
+    # Coexistence of long-form and short-form parameters.
+    with pytest.raises(
+        GMTInvalidInput,
+        match="Parameter in short-form 'J' conflicts with long-form parameter 'projection'.",
+    ):
+        func(projection="X10c", J="H10c")
+
+
+def test_alias_system_multiple_aliases_short_form():
+    """
+    Test that the alias system works with multiple aliases when short-form parameters
+    are used.
+    """
+    # Long-form exists but is not given, and short-form is given.
+    with pytest.warns(
+        SyntaxWarning,
+        match="Short-form parameter 'U' is not recommended. Use long-form parameter 'label', 'text' instead.",
```

Comment:
This error message might be confusing. So `label` is an alias for `-U`, and `text` is an alias for the `+t` modifier. Could we improve the message to something like:

```
Short-form parameter 'U' is not recommended. Use long-form parameter 'label' instead, with optional 'text' (+t) modifier.
```

Or something that makes it clearer that `U=label` and `+t=text`.

Reply:
I've improved the error message in 76f7c6ee125aad2ec9ea5254efc97e86c1a94286.

Now the messages are:

```
Short-form parameter 'U' is not recommended. Use long-form parameters 'label', with optional parameters 'text' (+t) instead.
```
and
```
Short-form parameter 'U' conflicts with long-form parameters and is not recommended. Use long-form parameters 'label', with optional parameters 'text' (+t) instead.
```